app/src/main/kotlin/com/wire/android/util/UriUtil.kt

```diff
@@ -0,0 +1,36 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.android.util
+
+import java.net.URLDecoder
+
+fun containsSchema(url: String): Boolean {
+    val regexPattern = Regex(".+://.+")
+
+    return regexPattern.matches(url)
```

Comment:
Suggestion: leverage already existing functions in the JVM.

```suggestion
    return URI.create(string).schema != null
```

Reply:
this only validates against valid and known schemes, it's not passing against unknown schemes.

Reply:
@mchenani how so?

<img width="960" alt="image" src="https://github.com/wireapp/wire-android-reloaded/assets/9389043/229057d1-7e1d-4975-969b-ee3ae90eaad7">